### PR TITLE
core/execute: return ENOSYS for unknown syscalls

### PR DESCRIPTION
**Pull request by koalo**
C.f. #19922.

### CODE REVIEW

**Comment by poettering** [comment]
https://systemd.io/CODING_STYLE/#formatting

**Comment by koalo** [comment]
whoopsie, sorry, should be fixed now

**Comment by bluca** [comment]
This seems to cause most tests to fail

**Comment by koalo** [comment]
I don't get it. If I understand the logs correctly, systemd-journald.service and others do not start. But all unit tests including the one that executes test/test-execute/exec-systemcallfilter-with-errno-in-allow-list.service can be executed successfully. Also all services start properly on my device with this MR. There is something I overlook but I am too unexperienced with the systemd internals to see it. Maybe it would have been better to just file the issue instead of trying to fix it on my own :-(

**Comment by mrc0mmand** [comment]
There seems to be some kind of performance issue:

```
[  OK  ] Finished Load/Save Random Seed.
[   37.379361] audit: type=1130 audit(1626871984.294:93): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=systemd-random-seed comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
[   37.379418] systemd[1]: systemd-random-seed.service: Control group is empty.
[   37.388439] systemd[1]: first-boot-complete.target: ConditionFirstBoot=yes failed.
[   37.390849] systemd[1]: first-boot-complete.target: Starting requested but condition failed. Not starting unit.
[   37.393972] systemd[1]: first-boot-complete.target: Job 142 first-boot-complete.target/start finished, result=done
[   37.397165] systemd[1]: Condition check resulted in First Boot Complete being skipped.
[   37.399614] systemd[1]: systemd-boot-system-token.service: starting held back, waiting for: local-fs.target
[   37.402637] systemd[1]: systemd-machine-id-commit.service: starting held back, waiting for: local-fs.target
[   47.583059] systemd-tmpfiles[565]: 	/etc/tmpfiles.d
[   47.583128] systemd-sysctl[560]: Parsing /usr/lib/sysctl.d/50-pid-max.conf
[   57.823051] systemd-tmpfiles[565]: 	/run/tmpfiles.d
[   57.823096] systemd-sysctl[560]: Setting 'kernel/core_pattern' to '|/usr/lib/systemd/systemd-coredump %P %u %g %s %t %c %h'.
[   68.063071] systemd-tmpfiles[565]: 	/usr/local/lib/tmpfiles.d
[   68.063086] systemd-sysctl[560]: Setting 'kernel/core_pipe_limit' to '16'.
[   78.303054] systemd-tmpfiles[565]: 	/usr/lib/tmpfiles.d
[   78.303061] systemd-sysctl[560]: Setting 'fs/suid_dumpable' to '2'.
[   88.543104] systemd-tmpfiles[565]: SELinux enabled state cached to: disabled
[   88.543110] systemd-sysctl[560]: Setting 'kernel/sysrq' to '16'.
[   96.492195] systemd[1]: systemd-journald.service: start operation timed out. Terminating.
[   96.496147] systemd[1]: systemd-journald.service: Changed start -> stop-sigterm
[   96.499159] systemd[1]: Received SIGCHLD from PID 558 ((journald)).
[   96.501621] systemd[1]: Child 558 ((journald)) died (code=killed, status=15/TERM)
[   96.504558] systemd[1]: systemd-journald.service: Child 558 belongs to systemd-journald.service.
[   96.507920] systemd[1]: systemd-journald.service: Main process exited, code=killed, status=15/TERM (success)
[   96.511710] systemd[1]: systemd-journald.service: Failed with result 'timeout'.
[   96.514582] systemd[1]: systemd-journald.service: Service will restart (restart setting)
[   96.517653] systemd[1]: systemd-journald.service: Changed stop-sigterm -> failed
[   96.520795] systemd[1]: systemd-journald.service: Job 223 systemd-journald.service/start finished, result=failed
[   96.524732] systemd[1]: Failed to start Journal Service.
[FAILED] Failed to start Journal Service.
See 'systemctl status systemd-journald.service' for details.
[   96.529883] systemd[1]: systemd-journal-flush.service: Job 158 systemd-journal-flush.service/start finished, result=dependency
[   96.534240] systemd[1]: Dependency failed for Flush Journal to Persistent Storage.
[   96.537245] audit: type=1334 audit(1626872043.453:94): prog-id=26 op=UNLOAD
[DEPEND] Dependency failed for Flus…Journal to Persistent Storage.
[   96.542591] systemd[1]: systemd-journal-flush.service: Job systemd-journal-flush.service/start failed with result 'dependency'.
[   96.546709] systemd[1]: systemd-journald.socket: Changed running -> listening
[   96.549124] systemd[1]: systemd-journald-audit.socket: Changed running -> listening
[   96.551622] systemd[1]: systemd-journald.service: Unit entered failed state.
[   96.554075] systemd[1]: systemd-journald.service: Consumed 10ms CPU time.
[   96.554086] audit: type=1130 audit(1626872043.469:95): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=systemd-journald comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=failed'
[   96.556865] systemd[1]: systemd-journald.service: Changed failed -> auto-restart
[   96.568301] systemd[1]: systemd-journald.service: Control group is empty.
[   96.570916] systemd[1]: systemd-sysctl.service: start operation timed out. Terminating.
[   96.574680] systemd[1]: systemd-sysctl.service: Changed start -> stop-sigterm
[   96.577569] systemd[1]: Received SIGCHLD from PID 560 (systemd-sysctl).
[   96.580167] systemd[1]: Child 560 (systemd-sysctl) died (code=killed, status=15/TERM)
[   96.583003] systemd[1]: systemd-sysctl.service: Child 560 belongs to systemd-sysctl.service.
[   96.585663] systemd[1]: systemd-sysctl.service: Main process exited, code=killed, status=15/TERM
[   96.588461] systemd[1]: systemd-sysctl.service: Failed with result 'timeout'.
[   96.590696] systemd[1]: systemd-sysctl.service: Service will not restart (restart setting)
[   96.593255] systemd[1]: systemd-sysctl.service: Changed stop-sigterm -> failed
[   96.595764] systemd[1]: systemd-sysctl.service: Job 166 systemd-sysctl.service/start finished, result=failed
[   96.598766] systemd[1]: Failed to start Apply Kernel Variables.
[FAILED] Failed to start Apply Kernel Variables.
See 'systemctl status systemd-sysctl.service' for details.
[   96.604818] systemd[1]: systemd-sysctl.service: Unit entered failed state.
[   96.607129] systemd[1]: systemd-sysctl.service: Consumed 43ms CPU time.

```

So far I haven't been able to find out what's going on (and the fact that the issue is not reproducible when `systemd.log_target=console` doesn't help either).

File: src/core/execute.c [hunk 1]
```diff
@@ -1471,7 +1471,8 @@ static bool skip_seccomp_unavailable(const Unit* u, const char* msg) {
 }
 
 static int apply_syscall_filter(const Unit* u, const ExecContext *c, bool needs_ambient_hack) {
-        uint32_t negative_action, default_action, action;
+        uint32_t default_action, action;
+        Hashmap *known;
```
Comment (poettering) [inline src/core/execute.c:1475]: please use `_cleanup_` handlers to destroy this after use

**Comment by poettering** [comment]
please fix the issues found by CI

**Comment by poettering** [comment]
let's close this for now, due to lack of response. if there's still interest, please address review+CI issues and file a new PR.

**Pull request closed without merging**